code/modules/mob/living/silicon/robot/inventory.dm

```diff
@@ -266,6 +266,30 @@
 	else
 		src << SPAN_NOTICE("You need to disable a module first!")
 
-/mob/living/silicon/robot/put_in_hands(var/obj/item/W) // No hands.
-	W.loc = get_turf(src)
-	return 1
+
+//Attempt to grip the item in a gripper.
+//Parent call will drop it on the floor if gripper can't hold it
+/mob/living/silicon/robot/put_in_hands(var/obj/item/W)
+	var/obj/item/weapon/gripper/G = null
+	if (istype(module_state_1, /obj/item/weapon/gripper))
```

Comment:
If you use
`var/obj/item/weapon/gripper/G = locate() in list(module_state_1, module_state_2, module_state_3)` (or loop)
there will be less copy-past.

Reply:
doing